### PR TITLE
docs : add rbc docs

### LINKED ISSUE #32: [rbc] writing docs
**Subject**
write here

**Detail**
write here

**Reference**
write here

### PR DESCRIPTION
**Pull request by AgwaB**
## Related Issue

resolve: #32 

## Description

add rbc docs

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] Test case
- [x] End of Work

### CODE REVIEW

File: docs/RBC_EN.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+HBBFT does not send the entire data but sends the pieces by dividing the data by the number of nodes in the network. For this purpose, RBC uses a technique called `erasure coding`.
+
+### Erasure coding   
+
+Erasure coding is one of FEC (Forward Error Correction) methods and it can perform error detection and error correction. In RBC, Reed-solomon technique is used among erasure coding techniques. It splits the original data into n pieces and adds k parity. Then, even if a maximum of k data is lost, original data can be recovered if only n data are survived. Good Examples and detailed explanations can be found [here](<https://www.backblaze.com/blog/reed-solomon/>).
+
+<p align="center">
+        <img src="../img/rbc_algorithm.png" width="450px" height="600px"></img>
```
Comment (hihiboss) [inline docs/RBC_EN.md]: The image size is `558 x 586`. So, how about `width="550px" height="580px"`?

File: docs/RBC_EN.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# RBC : Reliable Broadcast
+
+
+
+### overview
```
Comment (hihiboss) [inline docs/RBC_EN.md]: Upper case! `Overview` :)

File: docs/RBC_EN.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+The typical Broadcast, which depends on implementation, does not guarantee message delivery and does not consider situations in which the sender is crash (the situation in which the message cannot be sent to all). However, the RBC guarantees message delivery and partially solves the situation where the sender crashes. And the Atomic Broadcast introduced in this [paper](<https://eprint.iacr.org/2016/199.pdf>) is a Broadcast that guarantees the situation of the sender crash and guarantees the order of messages, and the RBC to be introduced is Atomic Broadcast.
```
Comment (hihiboss) [inline docs/RBC_EN.md]: The Atomic Broadcast seems to be suddenly introduced, so I suggest this as follow!

```suggestion
The typical Broadcast, which depends on implementation, does not guarantee message delivery and does not consider situations in which the sender is crash (the situation in which the message cannot be sent to all). However, the RBC guarantees message delivery and partially solves the situation where the sender crashes. 



In this [paper](<https://eprint.iacr.org/2016/199.pdf>), the Atomic Broadcast is introduced. This is a Broadcast that guarantees the situation of the sender crash and the order of messages. And this Atomic Broadcast is called the **RBC**.
```

File: docs/RBC_EN.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+
+[The paper](<https://eprint.iacr.org/2016/199.pdf>) shows ABC (Atomic Broadcast) as ACS (Asynchronous Common Subset) using `threshold encryption`and ACS consists of RBC and BBA (Binary Byzantine Agreement) components. Among them, RBC performs reliable broadcasting and BBA is responsible for ordering of consensus and atomic messages.
```
Comment (hihiboss) [inline docs/RBC_EN.md]: Typo! Plz space between `threshold encryption` and `and`.

Comment (hihiboss) [reply docs/RBC_EN.md]: In `responsible for ordering of consensus and atomic messages`, `ordering of consensus` is confused to me. Did you intend `ordering of consensus` **meaning a sequence of consensus**, not **consensus for ordering of atomic messages**?

File: docs/RBC_EN.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+
+HBBFT does not send the entire data but sends the pieces by dividing the data by the number of nodes in the network. For this purpose, RBC uses a technique called `erasure coding`.
+
+### Erasure coding   
+
+Erasure coding is one of FEC (Forward Error Correction) methods and it can perform error detection and error correction. In RBC, Reed-solomon technique is used among erasure coding techniques. It splits the original data into n pieces and adds k parity. Then, even if a maximum of k data is lost, original data can be recovered if only n data are survived. Good Examples and detailed explanations can be found [here](<https://www.backblaze.com/blog/reed-solomon/>).
```
Comment (hihiboss) [inline docs/RBC_EN.md]: In `Then, even if a maximum of k data is lost, original data can be recovered if only n data are survived.`, is it right `n pieces`? I think that it should be `(n-k) pieces`.

File: docs/RBC_EN.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
```
Comment (hihiboss) [inline docs/RBC_EN.md:20]: At here, is it necessary a title like `### RBC` or `### RBC Algorithm`

File: docs/RBC_EN.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        <img src="../img/rbc_algorithm.png" width="450px" height="600px"></img>
+</p>
+
+
+Assume that the content to be agreed on the network is Block.
+
+- There are N nodes on the network.
+
+- Split a block into N pieces. And erasure coding to generate N-2f parity.
+- S(j) is a fragment and the message to be delivered consists of VAL (h, b(j), s(j)). h: Merkle tree root of S(j), b(j): Merkle tree branch of S(j)
```
Comment (hihiboss) [inline docs/RBC_EN.md]:
```suggestion
- The message to be delivered consists of `VAL(h, b(j), s(j))`. (h: Merkle tree root of s(j), b(j): Merkle tree branch of s(j), s(j): A fragment)
```

File: docs/RBC_EN.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+
+
+1. When receiving VAL (h, b(j), s(j)) to P(sender), it multicasts ECHO (h, b(j), s(j)) to the node except the sender and itself.
```
Comment (hihiboss) [inline docs/RBC_EN.md]: In `When receiving VAL (h, b(j), s(j)) to P(sender)`, `to` should changed to `from`.

**Review by hihiboss — COMMENTED** [review]
Please consider my comments :)

File: docs/RBC_EN.md (hunk 5 above)
Comment (hihiboss) [reply docs/RBC_EN.md]: @AgwaB 
Sorry, I was confused. You're right :)

File: docs/RBC_EN.md (hunk 1 above)
Comment (zeroFruit) [reply docs/RBC_EN.md]: agree. image is too narrow

File: docs/RBC_EN.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+2. When ECHO (h, b(j), s(j)) is received from P(j), check whether h is valid using merkle information s(j) and b(j). If it is not valid, discard it.
+3. When receiving a valid ECHO from an N-f node
+   - Interpolate S(j) from N-2f leaves.
```
Comment (zeroFruit) [inline docs/RBC_EN.md]: plz, more description about *Interpolate*

Comment (zeroFruit) [reply docs/RBC_EN.md]: + S(j) -> `s(j)`

File: docs/RBC_EN.md (hunk 8 above)
Comment (zeroFruit) [reply docs/RBC_EN.md]: + where is the description about
> except the sender and itself. 

?

Comment (zeroFruit) [reply docs/RBC_EN.md]: and `VAL(h, b(j), s(j))` should be fixed to `VAL(h, b(i), s(i))`

File: docs/RBC_EN.md (hunk 9 above)
Comment (zeroFruit) [reply docs/RBC_EN.md]: > When receiving a valid ECHO from an N-f node

This means that receiving valid ECHO message **with same `h`** from distinct `N-f` node?

File: docs/RBC_EN.md (hunk 8 above)
Comment (zeroFruit) [reply docs/RBC_EN.md]: I think it would be better there's more description about `sender`

**Comment by zeroFruit** [comment]
overall, plz change the format if the character is equation into ``

for example:

i => `i`
s(j) => `s(j)`

File: docs/RBC_EN.md (hunk 4 above)
Comment (AgwaB) [reply docs/RBC_EN.md]: sequence of consensus!

File: docs/RBC_EN.md (hunk 8 above)
Comment (AgwaB) [reply docs/RBC_EN.md]: @zeroFruit sender and itself no need to receive messgae again. so paper descript as `multicast`. you think it needs additional description?

**Review by zeroFruit — APPROVED** [review]

**Pull request merged**